lib/modules/dosomething/dosomething_api/resources/reportback_resource.inc

```diff
@@ -149,10 +157,20 @@ function _reportback_resource_access($op) {
  * @param  bool    $random
  * @param  int     $page
  * @param  bool    $load_user
+ * @param  int     $reaction_user
  * @param  bool    $flagged
  * @return array
  */
-function _reportback_resource_index($ids, $campaigns, $status, $count, $random, $page, $load_user, $flagged, $runs) {
+function _reportback_resource_index($ids, $campaigns, $status, $count, $random, $page, $load_user, $reaction_user, $flagged, $runs) {
+  if ($reaction_user) {
+    $reaction_user = user_load($reaction_user);
+
+    if ($reaction_user) {
+      global $user;
+      $user = $reaction_user;
+    }
+  }
```

Comment:
Same same. Maybe put this as a function in the reportback module file, and then it's called in the respective `index()` methods.